csrc/preseg_passes/make_resharding_contiguous.cpp

```diff
@@ -53,6 +53,10 @@ bool validateMeshes(Fusion* fusion) {
 // behavior will be modified in the future with allocation and loop domain being
 // propagated independently.
 void setLoopAndAllocationDomain(TensorView* tv) {
+  if (tv->getLoopDomain() == tv->getLogicalDomain()) {
+    // Do nothing if sharding is on logical domain.
+    return;
+  }
```

Comment:
Reordering for logical domain split is affecting Overlap tests since they rely on positions of position of stream axis to be at the front in the output: https://github.com/NVIDIA/Fuser/blob/30cd9355d017b5837bfea08f49ad2eba6de59605/csrc/host_ir/lower.cpp#L490-L492